### PR TITLE
Cinder Volume API changes for adding support for secrets in the future

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
Many of the in-tree volume sources support specification of secret(s) for specific volumes. This support is not present in cinder currently. 

This PR adds just the changes in the API, so work can be done in the future in the external openstack controller manager.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -992,6 +992,40 @@ type CinderVolumeSource struct {
 	// the ReadOnly setting in VolumeMounts.
 	// +optional
 	ReadOnly bool
+	// Optional: points to a secret object that has enough information to connect
```
Comment (liggitt) [inline pkg/apis/core/types.go]: I would recommend leaving this more generic ("Optional: points to a secret object containing parameters used to connect to OpenStack") until the issues raised in https://github.com/kubernetes/kubernetes/pull/61295#discussion_r187440285 are resolved (which it sounds like will be done separately and possibly externally)

**Comment by liggitt** [comment]
structure matches other volume / persistentvolume sources and LGTM. the documentation shouldn't get ahead of the designed use of the secret parameters (whether that is in-tree or external), so make that more generic, then LGTM

**Comment by dims** [comment]
Thanks @liggitt updated the text as recommended.

File: pkg/apis/core/types.go [hunk 2]
```diff
@@ -992,6 +992,40 @@ type CinderVolumeSource struct {
...
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs". Implicitly inferred to be "ext4" if unspecified.
+	// +optional
+	FSType string
+	// Optional: Defaults to false (read/write). ReadOnly here will force
+	// the ReadOnly setting in VolumeMounts.
+	// +optional
+	ReadOnly bool
+	// Optional: points to a secret object containing parameters used to connect
+	// to OpenStack. The secret data can contain fields from the "[Global]" section
```
[the remark this answers is not in the document]
Comment (dims) [reply pkg/apis/core/types.go]: fixed. thanks!

**Pull request merged**